src/SFA.DAS.EmployerIncentives.Domain/ApprenticeshipIncentives/ApprenticeshipIncentive.cs

```diff
@@ -278,25 +283,37 @@ private void SetStartDateChangeOfCircumstance(DateTime startDate)
             }
         }
 
-        private void SetLearningStoppedChangeOfCircumstance(LearningStoppedStatus learningStoppedStatus)
+        private async Task SetLearningStoppedChangeOfCircumstance(LearningStoppedStatus learningStoppedStatus, ICollectionCalendarService collectionCalendarService)
         {
             if (learningStoppedStatus.LearningStopped && Model.Status != IncentiveStatus.Stopped)
             {
                 Model.Status = IncentiveStatus.Stopped;
+                await RemoveEarningsAfterStopDate(learningStoppedStatus.DateStopped.Value, collectionCalendarService);
                 AddEvent(new LearningStopped(
                     Model.Id,
                     learningStoppedStatus.DateStopped.Value));
             }
         }
 
-        public void SetHasPossibleChangeOfCircumstances(bool hasPossibleChangeOfCircumstances)
+        private async Task RemoveEarningsAfterStopDate(DateTime dateStopped, ICollectionCalendarService collectionCalendarService)
         {
-            Model.HasPossibleChangeOfCircumstances = hasPossibleChangeOfCircumstances;
+            var collectionCalendar = await collectionCalendarService.Get();
+
+            RemoveUnpaidEarnings(Model.PendingPaymentModels.Where(x => x.DueDate > dateStopped));
+            foreach (var paidPendingPayment in PendingPayments.Where(x => x.DueDate > dateStopped))
```

Comment:
Could we create a AddClawbacks(IEnumerable<PendingPaymentModel> pendingPayments) method similar to the RemoveUnpaidEarnings method?